src/plugins/tonegen/notification.c

```diff
@@ -0,0 +1,277 @@
+static void notif_play(struct ausrv *ausrv, int type, uint32_t vol, int dur);
+static void notif_stop(struct ausrv *ausrv, int kill_stream);
+
+static struct method  method_defs[] = {
+    {NOTIF_INTERFACE, "StartNotificationTone", "uiu", start_notif_tone},
+    {NOTIF_INTERFACE, "StopTone", "", stop_tone},
+    {NULL, NULL, NULL, NULL}
+};
+
+static void     *notif_props = NULL;
+static uint32_t  vol_scale   = 100;
+
+int notif_init(void)
+{
+    return 0;
+}
+
+int notif_create_ngfd(struct tonegend *t)
+{
+    (void) t;
+
+    return 0;
+}
+
+int notif_create(struct tonegend *tonegend)
+{
+    struct method *m;
+    int err;
+    int sts;
+
+    for (m = method_defs, err = 0;    m->memb != NULL;    m++) {
+        sts = dbusif_register_input_method(tonegend, m->intf, m->memb,
+                                           m->sig, m->func);
+
+        if (sts < 0) {
+            N_ERROR(LOG_CAT "%s(): Can't register D-Bus method '%s'",
+                      __FUNCTION__, m->memb);
+            err = -1;
+        }
+    }
+
+    return err;
+}
+
+
+static int start_notif_tone(DBusMessage *msg, struct tonegend *tonegend)
+{
+    struct ausrv *ausrv = tonegend->ausrv_ctx;
+    uint32_t      event;
+    int32_t       dbm0;
+    uint32_t      duration;
+    uint32_t      volume;
+    int           beeptype;
+    int           success;
+
+    success = dbus_message_get_args(msg, NULL,
+                                    DBUS_TYPE_UINT32, &event,
+                                    DBUS_TYPE_INT32 , &dbm0,
+                                    DBUS_TYPE_UINT32, &duration,
+                                    DBUS_TYPE_INVALID);
+
+    if (!success) {
+        N_ERROR(LOG_CAT "%s(): Can't parse arguments", __FUNCTION__);
+        return FALSE;
+    }
+
+    volume = linear_volume(dbm0);
+
+    
+
+    TRACE("%s(): event %u  volume %d dbm0 (%u) duration %u msec",
+          __FUNCTION__, event, dbm0, volume, duration);
+
+
+    switch (event) {
+    case 66:  beeptype = TONE_DIAL;      break;
+    case 72:  beeptype = TONE_BUSY;      break;
+    case 73:  beeptype = TONE_CONGEST;   break;
+    case 256: beeptype = TONE_RADIO_ACK; break;
+    case 257: beeptype = TONE_RADIO_NA;  break;
+    case 74:  beeptype = TONE_ERROR;     break;
+    case 79:  beeptype = TONE_WAIT;      break;
+    case 70:  beeptype = TONE_RING;      break;
+        
+    default:
+        N_ERROR(LOG_CAT "%s(): invalid event %d", __FUNCTION__, event);
+        return FALSE;
+    }
+    
+    notif_play(ausrv, beeptype, volume, duration * 1000);
+    return TRUE;
+}
+
+static int stop_tone(DBusMessage *msg, struct tonegend *tonegend)
+{
+    struct ausrv *ausrv = tonegend->ausrv_ctx;
+
+    (void)msg;
+
+    TRACE("%s()", __FUNCTION__);
+
+    notif_stop(ausrv, TRUE);
+
+    return TRUE;
+}
+
+/*
+ * This function maps the RFC4733 defined
+ * power level of 0dbm0 - -63dbm0
+ * to the linear range of 0 - 100
+ */
+static uint32_t linear_volume(int dbm0)
```

Comment:
This function already appears in `src/plugins/tonegen/rfc4733.c`, just slightly different -- only use one / merge those?